Core/Containers/DynamicsSelector.hpp

```diff
@@ -143,7 +140,10 @@ class DynamicsSelector : public DynamicsBase<
             return nullptr;
         }
     }
-
+private:
+    // Helper function for retrieving the currently active model at runtime, by indexing the tuple
+    // using template recursion. Since all models inherit from the same base type under this class,
+    // their pointers can be casted to the base type
     template <size_t I = 0>
     const Base* GetActiveModel() const{
         if(I == index_){
```

Comment:
The `const` version of this function will probably also need to be in `public`

Reply:
Moved the private statement down to allow (models_ and index_) variables to be private, and const version of GetActiveModel() to be public.